### PR TITLE
refactor(share/shrex): improve shrex logging

### PR DESCRIPTION
**Pull request by walldiss**
## Overview
 - Previously shrex-nd and shrex-eds error logs were indistinguishable, that lead to confusion of errors source. 
 - add peerID to server logs
 - Improve some other shrex logging.
 

Resolves https://github.com/celestiaorg/celestia-node/issues/2035

### CODE REVIEW

File: share/p2p/peers/manager.go [hunk 1]
```diff
@@ -174,38 +174,49 @@ func (m *Manager) Stop(ctx context.Context) error {
 func (m *Manager) Peer(
 	ctx context.Context, datahash share.DataHash,
 ) (peer.ID, DoneFunc, error) {
+	logger := log.With("hash", datahash.String())
```
Comment (Wondertan) [inline share/p2p/peers/manager.go:177]: Any reason not to shadow the global logger by naming it `log`?

File: share/p2p/shrexeds/server.go [hunk 2]
```diff
@@ -90,44 +93,44 @@ func (s *Server) handleStream(stream network.Stream) {
...
 
 	err = stream.Close()
 	if err != nil {
-		log.Debugw("server: closing stream", "err", err)
+		logger.Debugw("server: closing stream", "err", err)
 	}
 }
 
-func (s *Server) readRequest(stream network.Stream) (*p2p_pb.EDSRequest, error) {
+func (s *Server) readRequest(logger *zap.SugaredLogger, stream network.Stream) (*p2p_pb.EDSRequest, error) {
```
Comment (renaynay) [inline share/p2p/shrexeds/server.go:130]: why?

File: share/p2p/shrexnd/client.go [hunk 3]
```diff
@@ -69,7 +69,7 @@ func (c *Client) RequestND(
 		}
 	}
 	if err != p2p.ErrNotFound {
-		log.Warnw("client-nd: peer returned err", "peer", peer, "err", err)
+		log.Warnw("client: peer returned err", "err", err)
```
Comment (renaynay) [inline share/p2p/shrexnd/client.go:72]: why remove nd?

File: share/p2p/shrexnd/server.go [hunk 4]
```diff
@@ -140,19 +143,19 @@ func validateRequest(req pb.GetSharesByNamespaceRequest) error {
 }
 
 // respondNotFoundError sends internal error response to client
-func (srv *Server) respondNotFoundError(stream network.Stream) {
+func (srv *Server) respondNotFoundError(logger *zap.SugaredLogger, stream network.Stream) {
```
Comment (renaynay) [inline share/p2p/shrexnd/server.go:146]: why

File: share/p2p/shrexnd/server.go [hunk 5]
```diff
@@ -179,20 +182,20 @@ func namespacedSharesToResponse(shares share.NamespacedShares) *pb.GetSharesByNa
 	}
 }
 
-func (srv *Server) respond(stream network.Stream, resp *pb.GetSharesByNamespaceResponse) {
+func (srv *Server) respond(logger *zap.SugaredLogger, stream network.Stream, resp *pb.GetSharesByNamespaceResponse) {
```
Comment (renaynay) [inline share/p2p/shrexnd/server.go:185]: i don't get why it needs to be passed

File: share/p2p/shrexeds/server.go (hunk 2 above)
Comment (walldiss) [reply share/p2p/shrexeds/server.go:130]: I want to use logger enriched with fields

**Review by renaynay — APPROVED** [review]

File: share/p2p/shrexnd/server.go (hunk 4 above)
Comment (renaynay) [reply share/p2p/shrexnd/server.go:146]: wait i still don't understand why it needs to be passed @walldiss

File: share/p2p/peers/manager.go (hunk 1 above)
Comment (Wondertan) [reply share/p2p/peers/manager.go:177]: Ok

**Review by Wondertan — APPROVED** [review]

File: share/p2p/shrexnd/server.go (hunk 4 above)
Comment (walldiss) [reply share/p2p/shrexnd/server.go:146]: I want to use logger with fields down the stack. Do you mean it could be done differently?

**Pull request merged**